src/ftest.jl

```diff
@@ -0,0 +1,92 @@
+"""A helper function to determine if mod1 is nested in mod2"""
+function issubmodel(mod1::LinPredModel, mod2::LinPredModel)
+    mod1.rr.y != mod2.rr.y && return false # Response variables must be equal
+
+    # Now, test that all predictor variables are equal
+    pred1 = mod1.pp.X
+    npreds1 = size(pred1, 2)
+    pred2 = mod2.pp.X
+    npreds2 = size(pred1, 2)
+    # If model 1 has more predictors, it can't possibly be a submodel
+    npreds1 > npreds2 && return false 
+    
+    @inbounds for i in 1:npreds1
+        var_in_mod2 = false
+        for j in 1:npreds2
+            if view(pred1, :, i) == view(pred2, :, j)
+                var_in_mod2 = true
+                break
+            end
+        end
+        if !var_in_mod2
+            # We have found a predictor variable in model 1 that is not in model 2
+            return false 
+        end
+    end
+
+    return true
+end
+
+"""
+    ftest(mod::LinPredModel...)
+
+For each sequential pair of linear predictors in `mod`, perform an F-test to determine if the first one fits significantly better than the second
+
+Note: This function can easily be used to do an ANOVA. ANOVA is nothing more than a test
+to see if one model fits the data better than another.
+
+# Examples:
+As usual, we want to compare a result across two or more treatments. In the classic ANOVA
```

Comment:
"usual" probably does not apply to all potential users. Let's remain factual here and just present the way of doing an ANOVA. Also "perfectly good fit" is a bit casual, maybe just "fits the data"?

In the note above, also better remain more neutral and just state that this function can be used to perform an ANOVA by comparing etc.